test/cases/uri_gid_test.rb

```diff
@@ -45,6 +45,28 @@ class URI::GIDTest <  ActiveSupport::TestCase
   end
 end
 
+class URI::GIDIDEncodingTest < ActiveSupport::TestCase
+  test 'alphanumeric' do
+    model = Person.new('John123')
+    assert_equal 'gid://app/Person/John123', URI::GID.create('app', model).to_s
+  end
+
+  test 'non-alphanumeric' do
+    model = Person.new('John Doe-Smith/Jones')
```

Comment:
I don't think I really matters if an id feels like an id. It matters if:

* It's accepted by other parts of the system, eg: `ActiveRecord`.
* That it's useful and occurs in the real wold.

Ultimately all and id has to do is identify, eg: be unique. Numeric ids are one thing, so are uuids, and there are infinately more legitimate scemes.